includes/class-sensei-learner.php

```diff
@@ -9,13 +9,70 @@
  * @since 1.9.0
  */
 class Sensei_Learner {
+	/**
+	 * Instance of singleton.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
 	/**
 	 * Cache of the learner terms.
 	 *
 	 * @var WP_Term[]
 	 */
 	private static $learner_terms = [];
 
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Sensei_Course_Enrolment_Manager constructor. Private so it can only be initialized internally.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Sets the actions.
+	 *
+	 * @since 3.0.0
+	 */
+	public function init() {
+		// Delete user activity and enrolment terms when user is deleted.
+		add_action( 'deleted_user', array( $this, 'delete_user_registers' ) );
+	}
+
+	/**
+	 * Delete all user activity and enrolment terms when user is deleted.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @param  integer $user_id User ID.
+	 * @return void
+	 */
+	public function delete_user_registers( $user_id = 0 ) {
```

Comment:
I think that `delete_user_activity` or `delete_all_user_activity` is a better name for this and it is totally fine to use it if we deprecate the one from Utils.

Reply:
I had renamed it because on the code the activity is being related to the activities save in the comments. But it also works for me :)
I renamed here: https://github.com/Automattic/sensei/commit/fcec99649d51c0b1d1e3c4689733829be1aa8892

Reply:
Oops! I did it, but I gave up because of this: https://github.com/Automattic/sensei/pull/2898#discussion_r379564193

I preferred to keep the methods separated after the deprecation to allow the users to use the original behavior if desired. What do you think? And if you prefer to give another name, or use another approach, suggestions are welcome. :)

Reply:
I can't think of scenario where someone would want to delete the user and keep the learner meta so I think that we  should really keep a single function called `delete_all_user_activity` in this class. What do you think @jom? Since Renatho is on leave I could possible implement this.

Reply:
I tend to agree that this could be a single function. I'm not sure that `registers` is the proper word choice here regardless.

Reply:
Thank you for the suggestion, folks! Refactored here: https://github.com/Automattic/sensei/pull/2898/commits/b969b0af5dd94fcaa1d1924f2761b62542e14a6d